doc/developers/contributing.rst

```diff
@@ -230,109 +234,456 @@ contribution must conform to the project's :ref:`coding guidelines
+          and run the downloaded `vs_buildtools.exe` file. During the installation you will
+          need to make sure you select "Desktop development with C++", similarly to this
+          screenshot:
 
-        pip install pytest pytest-cov ruff==0.11.2 mypy numpydoc
+          .. image::
+            ../images/visual-studio-build-tools-selection.png
 
-.. _upstream:
+          Next, install the 64-bit version of Python (|PythonMinVersion| or later), for instance from the
+          `official website <https://www.python.org/downloads/windows/>`__.
 
-6. Add the ``upstream`` remote. This saves a reference to the main
-   scikit-learn repository, which you can use to keep your repository
-   synchronized with the latest changes:
+          Now create a virtual environment (venv_) and install the required python packages:
 
-   .. prompt:: bash
+          .. prompt:: powershell
 
-        git remote add upstream https://github.com/scikit-learn/scikit-learn.git
+            python -m venv sklearn-dev
+            sklearn-dev\Scripts\activate  # activate
+            pip install wheel numpy scipy cython meson-python ninja \
+              pytest pytest-cov ruff==0.11.2 mypy numpydoc
 
-7. Check that the `upstream` and `origin` remote aliases are configured correctly
-   by running:
+        .. tab-item:: conda
+          :class-label: tab-6
+          :sync: package-manager-conda
 
-   .. prompt:: bash
+          First, you need to install a compiler with OpenMP_ support.
+          Download the `Build Tools for Visual Studio installer <https://aka.ms/vs/17/release/vs_buildtools.exe>`_
+          and run the downloaded `vs_buildtools.exe` file. During the installation you will
+          need to make sure you select "Desktop development with C++", similarly to this
+          screenshot:
+
+          .. image::
+            ../images/visual-studio-build-tools-selection.png
+
+          Next, install a recent version of Python (|PythonMinVersion| or later) for instance
+          using conda-forge_. Conda-forge provides a conda-based distribution of
+          Python and the most popular scientific libraries.
+          Create a new conda environment with the required python packages:
+
+          .. prompt:: powershell
+
+            conda create -n sklearn-dev -c conda-forge \
+              python numpy scipy cython meson-python ninja \
+              pytest pytest-cov ruff==0.11.2 mypy numpydoc
+
+          It is not always necessary but it is safer to open a new prompt before
+          activating the newly created conda environment:
+
+          .. prompt:: powershell
+
+            conda activate sklearn-dev
+
+    .. tab-item:: MacOS
+      :class-label: tab-4
+
+      .. tab-set::
+        :class: tabs-package-manager
+
+        .. tab-item:: pip
+          :class-label: tab-6
+          :sync: package-manager-pip
+
+          The default C compiler on macOS, Apple clang (confusingly aliased as
```

Comment:
I would either remove this (newcomers don't need to know this) or merge it with "Next, install the LLVM OpenMP library with Homebrew".

In any case, the confusingly aliased as `/usr/bin/gcc` feels like a maintainer remark, not something that is meaningful for a newcomer.